### PR TITLE
Setup markdownlint

### PR DESCRIPTION
**Pull request by buberdds**
- task that lint docs in repo
- `markdownlint` will be used to lint change log fragments too in a next PR.

Let me know if we want to keep `MD033` override in config or not. It is used to render collapsible section in Explorer readme.

### CODE REVIEW

**Review by csillag — APPROVED** [review]
Reviewed. LGTM.

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "lint": "eslint --ext js,ts,tsx ./",
     "lint:fix": "yarn run lint --fix",
     "gitlint": "node ./internals/scripts/gitlint.js",
+    "docslint": "markdownlint --ignore '**/node_modules/**' '**/*.md'",
```
Comment (tjanez) [inline package.json:20]: For command name, I would rather use the same style as "build", "build-preview", "build-internal", so:
"lint", "lint-git", "lint-docs".

File: README.md [hunk 2]
```diff
@@ -7,21 +7,25 @@
...
 </summary>
 
-OpenAPI spec for the bleeding-edge version is at https://github.com/oasisprotocol/oasis-indexer/blob/main/api/spec/v1.yaml.
+OpenAPI spec for the [bleeding-edge version][indexer-bleeding-edge-spec].
 
-To quickly run latest emerald indexer locally without running a node, replace https://github.com/oasisprotocol/oasis-indexer/blob/d48de37/tests/e2e/config/e2e-dev.yml#L1-L8 with:
+To quickly run latest emerald indexer locally without running a node,
+[replace following lines][replace-lines-to-run-latest-emerald] with:
 
+<!-- markdownlint-disable MD013 -->
```
Comment (tjanez) [inline README.md:28]: Similarly, I would prefer human readable names in the ignores so it is more obvious what they do and why they are there:

```suggestion
<!-- markdownlint-disable line-length -->
```

File: README.md [hunk 3]
```diff
@@ -34,6 +38,7 @@ analysis:
       to: <latest round>
       from: <latest round - 500>
 ```
+<!-- markdownlint-disable MD013 -->
```
Comment (tjanez) [inline README.md:41]: At the end, the same rule needs to be re-enabled:

```suggestion
<!-- markdownlint-enable MD013 -->
```

**Review by tjanez — APPROVED** [review]

**Pull request merged**